packages/contracts-core/contracts/manager/BondingManager.sol

```diff
@@ -88,8 +91,8 @@ contract BondingManager is AgentManager, InterfaceBondingManager {
         if (status.flag == AgentFlag.Unknown) {
             // Unknown address could be added to any domain
             // New agent will need to be added to `_agents` list: could not have more than 2**32 agents
-            if (_agents.length >= type(uint32).max) revert MerkleTreeFull();
```

Comment:
It's not removed, but rather the revert message is changed (now the generic message from `SafeCast` is used)

Reply:
In practice I don't think it's possible to have more than `2**32` historical agents, though it's a theorecically possible over-time DoS attack, just need to determine how much time it will take for the attacker to constantly stake-unstake using a fresh address every time (and taking one agent index at a time).

- This is tracked in #1429

Reply:
Awesome, let's address there if at all.